command/agent/auth/auth.go

```diff
@@ -77,6 +80,21 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) {
 	}()
 
 	credCh := am.NewCreds()
+	if !ah.enableReauthOnNewCredentials {
+		realCredCh := credCh
+		credCh = nil
+		if realCredCh != nil {
+			go func() {
+				for {
+					select {
+					case <-ctx.Done():
+						return
+					case <-realCredCh:
+					}
+				}
+			}()
+		}
+	}
 	if credCh == nil {
 		credCh = make(chan struct{})
```

Comment:
Is this needed?

Reply:
Can you be more specific? Is which part needed?

Reply:
Checking `credCh` for nil and assigning a new channel if it is nil. The nil channel will block, so I think the behavior is the same without this `if` block, right?